File: Java/300-400/AR_329.java
import java.io.*;
import java.util.*;

public class AR_329 {
    static class Pair implements Comparable<Pair> {
        int first, second;

        Pair(int first, int second) {
            this.first  =first;
            this.second =second;
        }

        @Override
        public int compareTo(Pair other) {
            return Integer.compare(this.first, other.first);
        }
    }

    static int voraz(List<Pair> edificios) {
        int ret=1;
        int fin=edificios.get(0).second;

        for (int i=1;i<edificios.size();i++) {
            if (edificios.get(i).first < fin) {
                if (fin>edificios.get(i).second) fin=edificios.get(i).second;
            } else {
                ret++;
                fin=edificios.get(i).second;
            }
        }
        return ret;
    }

    static boolean resuelveCaso(BufferedReader br, BufferedWriter bw) throws IOException {
        int N=Integer.parseInt(br.readLine().trim());
        if (N==0) return false;

        List<Pair> edificios=new ArrayList<>();
        for (int i=0;i<N;i++) {
            String[] parts=br.readLine().trim().split("\\s+");
            int ini=Integer.parseInt(parts[0]);
            int fin=Integer.parseInt(parts[1]);

            edificios.add(new Pair(ini, fin));
        }
        Collections.sort(edificios);

        bw.write(String.valueOf(voraz(edificios)));
        bw.newLine();

        return true;
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br=new BufferedReader(new InputStreambr(System.in));
        BufferedWriter bw=new BufferedWriter(new OutputStreambw(System.out));

        while (resuelveCaso(br, bw));
        bw.flush();        
    }
}

File: C/300-400/AR_329.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int first;
    int second;
} Pair;

int comparePairs(const void *a, const void *b) {
    Pair *pairA=(Pair *)a;
    Pair *pairB=(Pair *)b;

    return (pairA->first - pairB->first);
}

int voraz(Pair edificios[], int n) {
    int i;

    int ret=1;
    int fin=edificios[0].second;

    for (i=1;i<n;i++) {
        if (edificios[i].first<fin) {
            if (fin>edificios[i].second) fin=edificios[i].second;
        } 
        else {
            ret++;
            fin=edificios[i].second;
        }
    }

    return ret;
}

int resuelveCaso() {
    int i;

    int N;
    if (scanf("%d", &N)!=1 || N==0) return 0;

    Pair edificios[N];
    for (i=0;i<N;i++) {
        scanf("%d %d", &edificios[i].first, &edificios[i].second);
    }

    qsort(edificios, N, sizeof(Pair), comparePairs);

    printf("%d\n", voraz(edificios, N));
    return 1;
}

int main() {
    while (resuelveCaso());
    
    return 0;
}
